src/app/data-model-provider/MetadataTypes.h

```diff
@@ -84,18 +85,102 @@ enum class AttributeQualityFlags : uint32_t
     kTimed           = 0x0040, // `T` quality on attributes (writes require timed interactions)
 };
 
+
+struct attribute_entry_mask_t {
+
+    // attribute quality flags
+    //
+    std::underlying_type_t<AttributeQualityFlags> flags : 7 ;  // generally defaults to View if readable
+
+    // read/write access privilege variables
+    //
+    std::underlying_type_t<Access::Privilege> readPrivilege : 5 ;  // generally defaults to View if readable
+    std::underlying_type_t<Access::Privilege> writePrivilege : 5 ; // generally defaults to Operate if writable
+
+};
+
+
+// Static ASSERT to check size of mask type.
+static_assert(sizeof(attribute_entry_mask_t) <= 4, "Size of attribute_entry_mask_t is not as expected.");
+
+// zero is not a valid privilege, just a default initial value for privileges
+constexpr std::underlying_type_t<Access::Privilege> kNoPrivilege(0) ;
+
+
 struct AttributeEntry
 {
     AttributeId attributeId;
-    BitFlags<AttributeQualityFlags> flags;
 
-    // read/write access will be missing if read/write is NOT allowed
-    //
-    // NOTE: this should be compacted for size
-    std::optional<Access::Privilege> readPrivilege;  // generally defaults to View if readable
-    std::optional<Access::Privilege> writePrivilege; // generally defaults to Operate if writable
+    // Constructor
+    constexpr AttributeEntry() : attributeId(0), mask{ 0, kNoPrivilege, kNoPrivilege } {}
```

Comment:
we need a constexpr constructor that also allows us to initialize this to anything. We need to be able to create const arrays in flash of these entries (so that we do not consume RAM to alter them after the fact).

Please add a constructor that takes attribute + flags + privileges